code/game/objects/items/devices/instruments.dm

```diff
@@ -268,3 +268,37 @@
 	item_state = "electronic"
 	lefthand_file = 'icons/mob/inhands/misc/devices_lefthand.dmi'
 	righthand_file = 'icons/mob/inhands/misc/devices_righthand.dmi'
+
+/obj/item/musicbeacon
+	name = "express delivery beacon"
+	desc = "Summon your tool of art."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "gangtool-red"
+	item_state = "radio"
+
+/obj/item/musicbeacon/attack_self(mob/user)
+	beacon_music(user)
+
+/obj/item/musicbeacon/proc/beacon_music(mob/M)
+	var/list/instruments = list(
+								/obj/item/instrument/violin,
+								/obj/item/instrument/piano_synth,
+								/obj/item/instrument/guitar,
+								/obj/item/instrument/eguitar,
+								/obj/item/instrument/glockenspiel,
+								/obj/item/instrument/accordion,
+								/obj/item/instrument/trumpet,
+								/obj/item/instrument/saxophone,
+								/obj/item/instrument/trombone,
+								/obj/item/instrument/recorder,
+								/obj/item/instrument/harmonica
+								)
+	var/static/list/display_names = list()
+	for(var/V in instruments)

```

Comment:
Making it static is the first step, you need to make the list only get created once if it doesn't exist. It will exist every other run of the proc.